src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -455,20 +505,56 @@ private static int preallocatedHashmapCapacity(int elementsToHold) {
         }
     }
 
-    static class NamedArgsAndClosure {
+    /**
+     * This class holds the argument map and optional body of the step that is to be invoked.
```

Comment:
Just some suggestions, and I'm not sure if my Javadoc syntax is 100% right so I would check it in your IDE before applying directly from GitHub (and maybe change `@code` to `@link` as well).

```suggestion
     * This class holds the argument map and optional body of the step that is to be invoked.
     *
     * <p>For steps that have complex argument types, like `checkout` (which takes an {@code SCM}), when users use symbol-based syntax with those arguments, an instance of this class is created as the result of `invokeDescribable` and returned to the Groovy program so that it can be passed to `invokeStep` later, so it must implement {@code Serializable}.
```

Reply:
added in d489f73f2d33ad7eecf0b99bbea2e1291d72a35f